src/fm/utils.cljc

```diff
@@ -328,21 +333,28 @@
   (if (empty? (meta args-form))
     (fn-form form-args)
 
-    (let [metadata (into {} (map meta-xf) (meta args-form))
-          bindings (interleave
-                    (map :fm.meta/sym  (vals metadata))
-                    (map :fm.meta/form (vals metadata)))
-          fn-form  (fn-form
-                    (merge
-                     form-args
-                     {:fm/metadata metadata}))
-          meta     (not-empty
-                    (zipmap
-                     (keys metadata)
-                     (map :fm.meta/sym (vals metadata))))]
-
-      `(let [~@bindings]
-         (with-meta ~fn-form ~meta)))))
+    (let [metadata              (into {} (map meta-xf) (meta args-form))
+          metadata-vals         (vals metadata)
+          metadata-key-bindings (map :fm.meta/sym metadata-vals)
```

Comment:
Extracted a few common things you were doing multiple times in this binding.

Reply:
I realize my knee jerk toward this comes down to the amount of novelty that appears in a given function or code snippet.

It sounds positively outrageous saying it out loud, but as I'm parsing code I haven't read in a long time, I can essentially just prune `(vals metadata)`, because more of it was written by Rich (i.e. tested by core), vs. `metadata-vals`, which is something new (even though it's literally identical and a few lines away in this case). When something goes wrong, there are fewer "leaves" to inspect—i.e. something must be going wrong with the computation of `metadata`, because `vals` definitely isn't broken (except when it is!). In cases where performance is a consideration, only performing the computation once would obviously take precedence, but this is all happening as part of expansion.

Ultimately if it works it works, just broadcasting my reaction to these changes.